File: Python/BOJ_9613_GCD 합.py
import sys

def gcd(x, y):
    a = max(x, y); b = min(x, y)
    while b != 0:
        temp = a
        a = b
        b = temp % a
    return a

case = int(sys.stdin.readline())
for i in range(case):
    table = list(map(int, sys.stdin.readline().split()))
    table = table[1:]

    gcd_total = 0
    for j in range(len(table)):
        for k in range(j + 1, len(table)):
            gcd_total += gcd(table[j], table[k])
    print(gcd_total)

File: C++/BOJ_9613_GCD 합.cpp
#include <iostream>
#include <vector>

using namespace std;

int gcd(int x, int y)
{
    int temp;
    if (y > x){
        temp = x;
        x = y;
        y = temp;
    }
    while (x % y != 0) {
        temp = x % y;
        x = y;
        y = temp;
    }
    return y;
}

int main()
{
    int t;
    cin >> t;
    while (t--) {
        int n;
        long long sum = 0;
        cin >> n;
        vector <int> v(n);
        for (int i = 0; i < n; i++) {
            cin >> v[i];
        }
        
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                sum += gcd(v[i], v[j]);
            }
        }
        
        cout << sum << '\n';
    }
    return 0;
}